<!DOCTYPE html>
<!-- DO NOT EDIT! This test has been generated by /html/canvas/tools/gentest.py. -->
<meta charset="UTF-8">
<title>Canvas test: 2d.layer.valid-calls.save_reset_restore</h1><script><p>
</p></script><p class="desc">No exception raised on save() + reset() + restore().</p>


<p class="output">Actual output:</p>
<canvas id="c" class="output" width="100" height="50"><p class="fallback">FAIL (fallback content)</p></canvas>

<ul id="d"></ul><p class="fallback">
</p><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script><script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script><script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script><script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script><script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script><script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script><script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><ul id="d"><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script></ul><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctâ€­x) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script></script></script></script></script></script></script></script></script></script></script></script></script><script>
var t = async_test("No exception raised on save() + reset() + restore().");
_addTest(function(canvas, ctx) {

  ctx.save();
  ctx.reset();
  ctx.restore();

});
</script><script><script><script></script><script></script><meta charset="UTF-8"></script><script><script></script><script></script><meta charset="UTF-8"></script>

